### PR TITLE
Fixed some issues with master

### PR DESCRIPTION
**Pull request by mdproctor**
-For issue where PolyLine adjusts enhancement breaks OrthogonalLine, which does not impl the method and returns null.
-Fixed docking offsets, when the path BB does not have a minX and min…Y of 0.
-The snaps adjustDistance was the wrong way around, which gave weird behaviour while dragging a docked item.

### CODE REVIEW

**Comment by hasys** [comment]
Hi @mdproctor, @romartin,

I tested this PR with Rogers lienzo-test project. It looks like now, it is hard to redock node. Yes, behaviour while dragging a docked item is fixed, but if you want to redock it is had to find a point where you can do it.

Also there are missing tests for this changes to prevent regressions in the future.

File: src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresDockingControlImpl.java [hunk 1]
```diff
@@ -238,12 +238,13 @@ private boolean hasShapeOnMagnet(WiresMagnet magnet, WiresShape parent) {
...
                      final WiresContainer parent,
-                     final Point2D location) {
+                     final Point2D absLocation) {
         if (null != shape.getDockedTo()) {
             undock(shape, shape.getDockedTo());
         }
 
-        shape.setLocation(location);
+        Point2D relLocation = absLocation.sub(parent.getComputedLocation());
+        shape.setLocation(relLocation);
```
Comment (tiagodolphine) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresDockingControlImpl.java]: Need to set absLocation on the shape.

File: src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresDockingControlImpl.java [hunk 2]
```diff
@@ -257,18 +258,18 @@ public void dock(final WiresShape shape,
         }
 
         //adjust the location if necessary
-        final Point2D adjust = calculateCandidateLocation(shape, magnet);
-        if (!location.equals(adjust)) {
-            shape.setLocation(adjust);
+        final Point2D newAbsLocation = calculateAbsCandidateLocation(shape, magnet);
+        if (!absLocation.equals(newAbsLocation)) {
+            shape.setLocation(newAbsLocation.sub(parent.getComputedLocation()));
```
Comment (tiagodolphine) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresDockingControlImpl.java]: Need to set newAbsLocation on the shape.

**Comment by mdproctor** [comment]
more fixes, with squashed rebase and push.

**Comment by romartin** [comment]
Hey @hasys @tiagodolphine , please can you also verify Mark's new stuff? Thanks!

**Comment by hasys** [comment]
Hi @mdproctor, @romartin,

works good now, but tests are still missing.

**Comment by romartin** [comment]
Jenkins please retest this

**Comment by romartin** [comment]
Hey @mdproctor @hasys 

I tested and found a couple of issues:
* One test is failing
* On Stunner trying to undo/redo (docking) does not work correctly

**Comment by romartin** [comment]
@hasys The regular build does not fire the tests, which is not correct... let's try full
Jenkins please execute full downstream build

**Comment by romartin** [comment]
Hey @tiagodolphine 
You think can we close this PR and just reference the new one that includes all commits as well? 
https://github.com/kiegroup/lienzo-core/pull/52/
Thanks!

**Comment by hasys** [comment]
Hi @romartin,

in general it is easier to deal with smaller PRs, but if it is already merged to one mega PR I guess we have no other choose than close this PR.

**Comment by romartin** [comment]
@tiagodolphine  are you ok about closing this?

**Comment by tiagodolphine** [comment]
@romartin yes this could be closed since the changes referred by this PR will be merged on https://github.com/kiegroup/lienzo-core/pull/52. You can check the commits there.

**Pull request closed without merging**